pkg/sectigo/mock/mock.go

```diff
@@ -246,15 +246,63 @@ func (s *Server) createSingleCertBatch(c *gin.Context) {
 		c.JSON(http.StatusBadRequest, err)
 		return
 	}
-	c.JSON(http.StatusOK, &sectigo.BatchResponse{
-		BatchID:      42,
-		CreationDate: time.Now().Format(time.RFC3339),
-		Status:       sectigo.BatchStatusReadyForDownload,
-		Active:       false,
-		BatchName:    in.BatchName,
-		OrderNumber:  23,
-		Profile:      "profile",
-	})
+
+	subjectParams := map[string]bool{
+		"organizationName":    false,
+		"localityName":        false,
+		"stateOrProvinceName": false,
+		"countryName":         false,
+	}
+	nameParams := map[string]bool{
+		"commonName":     false,
+		"dNSName":        false,
+		"pkcs12Password": false,
+	}
+
+	for k, v := range in.ProfileParams {
+		if v == "" {
+			c.JSON(http.StatusBadRequest, fmt.Errorf("empty profile parameter %s", k))
+			return
+		}
+
+		if _, ok := subjectParams[k]; ok {
+			subjectParams[k] = true
+		} else if _, ok := nameParams[k]; ok {
+			nameParams[k] = true
+		} else {
+			c.JSON(http.StatusBadRequest, fmt.Errorf("invalid profile parameter %s", k))
+			return
+		}
+	}
+
+	hasSubject, hasName := true, true
+	for _, ok := range subjectParams {
+		if !ok {
+			hasSubject = false
+		}
+	}
+
+	for _, ok := range nameParams {
+		if !ok {
+			hasName = false
+		}
+	}
+
+	// ProfileCipherTraceEndEntityCertificate: includes subject and name parameters
+	// ProfileCipherTraceEE: only includes name parameters
+	if hasSubject && hasName || !hasSubject && hasName {
+		c.JSON(http.StatusOK, &sectigo.BatchResponse{
+			BatchID:      42,
+			CreationDate: time.Now().Format(time.RFC3339),
+			Status:       sectigo.BatchStatusReadyForDownload,
+			Active:       false,
+			BatchName:    in.BatchName,
+			OrderNumber:  23,
+			Profile:      "profile",
+		})
+	} else {
+		c.JSON(http.StatusBadRequest, fmt.Errorf("invalid profile parameters"))
+	}
```

Comment:
One of the input parameters is `AuthorityID` - which relates directly to the certificate profiles, e.g. when you call `sectigo.UserAuthorities` you get back something like this: 

```json
[
  {
    "id": 7,
    "ecosystemId": 18,
    "signerCertificateId": 0,
    "ecosystemName": "Testing",
    "balance": 25,
    "enabled": true,
    "profileId": 7,
    "profileName": "CipherTrace EE"
  }
]
```

`submitCertificateRequest` is doing this on [line 136](https://github.com/trisacrypto/directory/blob/main/pkg/gds/certs.go#L136) and then passing the `7` along with the request. 

What I propose is that if `AuthorityID` == 7; we test ProfileCipherTraceEE and if `AuthorityID == 24` or something we test the other profile. That way we can ensure based on the request what profile is returned. We'll probably also want to edit the `userAuthorities` handler and the `profiles`, `profileParameters`, and `profileDetail` endpoints to respond correctly based on the profile configured in the Mock … which is also something the mock might need to do.

Reply:
That makes sense, I was looking for something like that to make the mock handling logic easier but I couldn't find which Sectigo endpoint was doing it. I think we should create another story to configure the profile in the mock like you mentioned.